File: app/src/main/java/com/mulechat/app/data/PeerContact.java
package com.mulechat.app.data;

public final class PeerContact {
    public final String peerId;
    public final String displayName;
    public final byte[] ed25519PublicKey;
    public final byte[] x25519IdentityKey; // null until we've pinned this peer's first PreKeyBundle (see X3DH)
    public final String lastKnownHost; // null if never seen on the LAN
    public final Integer lastKnownPort; // null if never seen on the LAN

    public PeerContact(String peerId, String displayName, byte[] ed25519PublicKey, byte[] x25519IdentityKey,
                        String lastKnownHost, Integer lastKnownPort) {
        this.peerId = peerId;
        this.displayName = displayName;
        this.ed25519PublicKey = ed25519PublicKey;
        this.x25519IdentityKey = x25519IdentityKey;
        this.lastKnownHost = lastKnownHost;
        this.lastKnownPort = lastKnownPort;
    }
}

File: composeApp/src/commonMain/kotlin/com/nulchat/data/PeerContact.kt
package com.nulchat.data

data class PeerContact(
    val peerId: String,
    val displayName: String,
    val ed25519PublicKey: ByteArray,
    val lastKnownHost: String?,
    val lastKnownPort: Long?
)

data class DirectMessage(
    val id: String,
    val peerId: String,
    val body: String,
    val outgoing: Boolean,
    val sentAtEpochMs: Long,
    val deliveryState: String
)
